### ISSUE TITLE
A Simpler Idea - Is My Walk Safe?

### ISSUE THREAD

**Issue by bschoenfeld** [CodeforNRV/safe_walk_blacksburg#3]
It seems that we've taken on the challenge of finding the safest path, which is a pretty difficult problem. As an alternative, we could get the path from Google and just give it a safety score. [Here's a demo](http://www.geocodezip.com/v3_animate_marker_directions.html) where a marker is animated through a path. We could take a number of positions along the route, like that demo seems to do, and score them each. Then we could aggregate the scores and present a final score. We could then set thresholds and make recommendations (it's ok to walk, you should probably get a ride, etc).

If we do this, we just need to make a number of APIs that take a point and return a yes / no answer e.g. was there a recent crime here, is there a sidewalk here, etc.

Future iterations could show the dangerous portions of the walk on a map and allow the user to manually change the route and re-score.

**Comment by nealf** [comment]
Yes, I think we can definitely do this over several iterations, keeping things simple at first and then adding functionality. I envisioned the first release essentially just showing a map with something like green, yellow, and red roads along with start location and ending destination. I prefer that to doing a Google route with a safety score, as I could see that getting confusing for users ("This is the route it gave me, but it says it isn't safe? What direction am I supposed to go then?!") 

Then we can look at adding routing, then maybe route options, and other functionality and features.

**Comment by picaso66** [comment]
I just became aware of this project. Just a thought: would you consider adding areas located near registered sex offenders to be less safe? I know it would be irrelevant in many cases, but it might significant to some users.

**Comment by nealf** [comment]
That's an interesting idea, and definitely one we can consider implementing down the road. I don't think that ever came up in discussions with our police chief, which is why I don't think we had ever thought about it. I imagine we could add it as another optional overlay for those people who might find it significant, similar to how we're planning to do the crime layer. The bigger question is probably how that data is published (machine readable format? update frequency?). Definitely something we can look into, thanks for the suggestion!
